src/server.cpp

```diff
@@ -849,16 +863,18 @@ static CTimingMeas JitterMeas ( 1000, "test2.dat" ); JitterMeas.Measure(); // TE
                 const int iStartChanCnt = iBlockCnt * iMTBlockSize;
                 const int iStopChanCnt  = std::min ( ( iBlockCnt + 1 ) * iMTBlockSize - 1, iNumClients - 1 );
 
-                FutureSynchronizer.addFuture ( QtConcurrent::run ( this,
-                                                                   &CServer::DecodeReceiveDataBlocks,
-                                                                   iStartChanCnt,
-                                                                   iStopChanCnt,
-                                                                   iNumClients ) );
+                Futures.push_back ( pThreadPool->enqueue ( CServer::DecodeReceiveDataBlocks,
+                                                           this,
+                                                           iStartChanCnt,
+                                                           iStopChanCnt,
+                                                           iNumClients ) );
             }
 
             // make sure all concurrent run threads have finished when we leave this function
-            FutureSynchronizer.waitForFinished();
-            FutureSynchronizer.clearFutures();
+            for ( auto& future : Futures ) {
```

Comment:
I'm not a fan of `auto` - it means I have to think "so what does this actually mean" - it's not self-documenting.  Given `Futures` isn't defined locally, I think it would be nice to declare `future` explicitly.

Reply:
So is `auto` just a shorthand, or does it make something possible to express that can't otherwise be expressed succinctly?

Reply:
> I'm not a fan of `auto` - it means I have to think "so what does this actually mean" - it's not self-documenting. Given `Futures` isn't defined locally, I think it would be nice to declare `future` explicitly.

The use of `auto` seems to be a standard C++11 idiom to be able to declare the loop variable succinctly and in a DRY (don't repeat yourself) way. So I would strongly favour leaving these as they are.